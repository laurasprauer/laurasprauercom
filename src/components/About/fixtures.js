export default {
  default: {
    content: [
      {
        title: 'Resume',
        subTitle: 'skills & experience',
        type: 'Resume',
      },
    ],
  },
};
