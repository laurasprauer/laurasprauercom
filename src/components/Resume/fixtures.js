export default {
  default: {
    content: [
      {
        company: 'Joot',
      },
    ],
  },
};
