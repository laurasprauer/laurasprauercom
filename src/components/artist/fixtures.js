export default {
  default: {
    content: [
      {
        title: 'Art',
      },
    ],
  },
};
