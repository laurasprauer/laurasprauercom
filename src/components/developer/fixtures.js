export default {
  default: {
    slug: '/developer',
    title: 'Developer',
    content: [
      {
        title: 'Resume',
        subTitle: 'experience & skills',
        link: null,
        type: 'Resume',
        techTalkSlug: null,
        techTalkPdf: null,
        internal: {
          type: 'ContentfulResource',
        },
      },
      {
        title: 'About Me',
        subTitle: 'based on a true story',
        link: '/about',
        type: 'Internal Link',
        techTalkSlug: null,
        techTalkPdf: null,
        internal: {
          type: 'ContentfulResource',
        },
      },
      {
        title: 'Insights from a Joot Engineer',
        subTitle: 'navigating startup life',
        link: 'https://joot.io/blog/insights-from-a-joot-engineer-navigating-the-intricacies-of-startup-life',
        type: 'External Link',
        techTalkSlug: null,
        techTalkPdf: null,
        internal: {
          type: 'ContentfulResource',
        },
      },
      {
        title: 'JavaScript History',
        subTitle: 'momentum 2021 talk & article',
        link: 'https://www.ample.co/blog/javascript-history',
        type: 'Tech Talk External Link',
        techTalkSlug: '/momentum-2021',
        techTalkPdf: {
          file: {
            url: '//assets.ctfassets.net/okczg0bbqwnf/4WUs9IinkTKBGVnJHqY7DK/a176af7030c376c1ae45f26d8a7edf4b/25-Years-of-JavaScript-Momentum-2021.pdf',
          },
        },
        internal: {
          type: 'ContentfulResource',
        },
      },
      {
        title: 'What is Jamstack?',
        subTitle: 'interview with ample devs',
        link: 'https://www.ample.co/blog/what-is-jamstack-our-developers-explain-what-a-jamstack-website-is',
        type: 'External Link',
        techTalkSlug: null,
        techTalkPdf: null,
        internal: {
          type: 'ContentfulResource',
        },
      },
    ],
  },
};
