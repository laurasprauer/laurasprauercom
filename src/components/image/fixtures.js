export default {
  default: {
    src: {
      childImageSharp: {
        gatsbyImageData: {
          images: {
            sources: [
              {
                srcSet:
                  'https://images.ctfassets.net/okczg0bbqwnf/1a41f8AOyLn6My3j7LOatU/16c0818341693655a26d936f17bb8f0c/laura-img-5.jpg?w=358&h=358&q=50&fm=webp 358w,\nhttps://images.ctfassets.net/okczg0bbqwnf/1a41f8AOyLn6My3j7LOatU/16c0818341693655a26d936f17bb8f0c/laura-img-5.jpg?w=717&h=717&q=50&fm=webp 717w,\nhttps://images.ctfassets.net/okczg0bbqwnf/1a41f8AOyLn6My3j7LOatU/16c0818341693655a26d936f17bb8f0c/laura-img-5.jpg?w=1433&h=1433&q=50&fm=webp 1433w',
                sizes: '(min-width: 1433px) 1433px, 100vw',
                type: 'image/webp',
              },
            ],
            fallback: {
              src: 'https://images.ctfassets.net/okczg0bbqwnf/1a41f8AOyLn6My3j7LOatU/16c0818341693655a26d936f17bb8f0c/laura-img-5.jpg?w=1433&h=1433&fl=progressive&q=50&fm=jpg',
              srcSet:
                'https://images.ctfassets.net/okczg0bbqwnf/1a41f8AOyLn6My3j7LOatU/16c0818341693655a26d936f17bb8f0c/laura-img-5.jpg?w=358&h=358&fl=progressive&q=50&fm=jpg 358w,\nhttps://images.ctfassets.net/okczg0bbqwnf/1a41f8AOyLn6My3j7LOatU/16c0818341693655a26d936f17bb8f0c/laura-img-5.jpg?w=717&h=717&fl=progressive&q=50&fm=jpg 717w,\nhttps://images.ctfassets.net/okczg0bbqwnf/1a41f8AOyLn6My3j7LOatU/16c0818341693655a26d936f17bb8f0c/laura-img-5.jpg?w=1433&h=1433&fl=progressive&q=50&fm=jpg 1433w',
              sizes: '(min-width: 1433px) 1433px, 100vw',
            },
          },
          layout: 'constrained',
          backgroundColor: '#282828',
          width: 1433,
          height: 1433,
        },
      },
    },
  },
  native: {
    src: '//images.ctfassets.net/okczg0bbqwnf/1a41f8AOyLn6My3j7LOatU/16c0818341693655a26d936f17bb8f0c/laura-img-5.jpg',
    width: '750',
    height: '399',
  },
};
