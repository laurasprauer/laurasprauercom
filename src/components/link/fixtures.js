export default {
  internal: {
    to: '/about',
    children: "I'm an internal link",
  },
  external: {
    to: 'https://joot.io/',
    children: "I'm an external link",
  },
};
