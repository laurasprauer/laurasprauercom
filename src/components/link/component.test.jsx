import React from 'react';
import renderer from 'react-test-renderer';

import Link from './component';
import fixtures from './fixtures';

describe('Internal Link', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Link {...fixtures.internal} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('External Link', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Link {...fixtures.external} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
