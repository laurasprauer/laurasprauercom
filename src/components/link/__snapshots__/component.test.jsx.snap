// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`External Link renders correctly 1`] = `
<a
  className="
    undefined
    
    
  "
  href="https://joot.io/"
  rel=""
  target="_self"
>
  I'm an external link
</a>
`;

exports[`Internal Link renders correctly 1`] = `
<a
  className="
    undefined
    
    
  "
  href="/about"
  target="_self"
>
  I'm an internal link
</a>
`;
